src/pybind/mgr/dashboard/frontend/cypress/integration/cluster/hosts.po.ts

```diff
@@ -161,4 +161,21 @@ export class HostsPageHelper extends PageHelper {
         });
     }
   }
+
+  @PageHelper.restrictTo(pages.index.url)
+  drain(hostname: string) {
+    this.getTableCell(this.columnIndex.hostname, hostname).click();
+    this.clickActionButton('drain');
+    this.checkLabelExists(hostname, ['_no_schedule'], true);
+
+    this.clickTab('cd-host-details', hostname, 'Daemons');
+    cy.get('cd-host-details').within(() => {
+      // draining will take some time to complete.
+      // since we don't know how many daemons will be
+      // running in this host in future putting the wait
+      // to 20s
+      cy.wait(20000);
```

Comment:
```suggestion
      cy.wait(20000);
      this.getTableCount('total').should('be.eq', 0);
```
Instead of hardcoded waiting value, have you explored the option to [add a timeout ](https://docs.cypress.io/api/commands/should#Timeouts)to the chained previous command to `.should()` ?
That way the test should not wait 20s if the thing to check can be checked as soon as it's available: e.g. we could check the number of rows, the content of the appropriate column, ...

Reply:
disabled the test for now! i'll defer it for a later PR.

Reply:
I've kept a wait of 10s because removing an osd is taking its time and I saw the tests going into a timeout error.